File: LinkedInWebApi/Logic/Models/LinkedInUserData.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Logic.Models
{
    public class Identifier
    {
        public string identifier { get; set; }
        public int index { get; set; }
        public string mediaType { get; set; }
        public string file { get; set; }
        public string identifierType { get; set; }
        public int identifierExpiresInSeconds { get; set; }
    }

    public class Element
    {
        public List<Identifier> identifiers { get; set; }
    }

    public class DisplayImage
    {
        public List<Element> elements { get; set; }
    }

    public class ProfilePicture
    {
        [JsonProperty("displayImage~")]
        public DisplayImage DisplayImage { get; set; }
    }

    public class LinkedInUserData
    {
        [JsonProperty(PropertyName = "id")]
        public string id { get; set; }
        [JsonProperty(PropertyName = "localizedFirstName")]
        public string localizedFirstName { get; set; }
        [JsonProperty(PropertyName = "localizedLastName")]
        public string localizedLastName { get; set; }
        [JsonProperty(PropertyName = "localizedHeadline")]
        public string localizedHeadline { get; set; }
        [JsonProperty(PropertyName = "profilePicture")]
        public ProfilePicture profilePicture { get; set; }
    }
}

File: AngularLinkedIn/src/app/models/linkedin-user-data.ts
export interface Identifier {
  identifier: string;
  index: number;
  mediaType: string;
  file: string;
  identifierType: string;
  identifierExpiresInSeconds: number;
}

export interface IElement {
  identifiers: Identifier[];
}

export interface IDisplayImage {
  elements: IElement[];
}

export interface IProfilePicture {
  displayImage: IDisplayImage;
}

export interface ILinkedInUserData {
  id: string;
  localizedFirstName: string;
  localizedLastName: string;
  profilePicture: IProfilePicture;
}

